### DISCUSSION TITLE
TypeError: Cannot read property 'prototype' of undefined, js engine: hermes

### DISCUSSION THREAD

**Discussion by rickyanwar** [software-mansion/react-native-reanimated#5155] [Q&A]
after upgrade expo project , now if i open my component have react reanimated always show this error
```

 ERROR  TypeError: Cannot read property 'prototype' of undefined, js engine: hermes
 ERROR  Invariant Violation: "main" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes

```
version i used
{
  "name": "t_horizon",
  "version": "1.0.0",
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo run:android",
    "ios": "expo run:ios",
    "web": "expo start --web"
  },
  "dependencies": {
    "@babel/preset-typescript": "^7.17.12",
    "@ptomasroos/react-native-multi-slider": "^2.2.2",
    "@react-native-async-storage/async-storage": "1.18.2",
    "@react-native-community/clipboard": "^1.5.1",
    "@react-native-community/datetimepicker": "7.2.0",
    "@react-native-firebase/app": "^18.5.0",
    "@react-native-firebase/messaging": "^18.5.0",
    "@react-native-picker/picker": "2.4.10",
    "@react-navigation/drawer": "^5.12.5",
    "@react-navigation/native": "^5.9.4",
    "@react-navigation/stack": "^5.14.5",
    "@stripe/stripe-react-native": "0.28.0",
    "@twotalltotems/react-native-otp-input": "^1.3.5",
    "axios": "^0.27.2",
    "expo": "^49.0.0",
    "expo-app-loading": "~2.0.0",
    "expo-av": "~13.4.1",
    "expo-community-flipper": "^45.1.0",
    "expo-dev-client": "~2.4.11",
    "expo-device": "~5.4.0",
    "expo-file-system": "~15.4.4",
    "expo-image-picker": "~14.3.2",
    "expo-linear-gradient": "~12.3.0",
    "expo-location": "~16.1.0",
    "expo-notifications": "~0.20.1",
    "expo-screen-capture": "~5.3.0",
    "expo-speech": "~11.3.0",
    "expo-splash-screen": "~0.20.5",
    "expo-status-bar": "~1.6.0",
    "expo-task-manager": "~11.3.0",
    "expo-updates": "~0.18.14",
    "firebase": "^10.4.0",
    "formik": "^2.4.3",
    "geolib": "^3.3.3",
    "i18next": "^22.0.4",
    "lodash": "^4.17.21",
    "moment": "^2.29.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-i18next": "^12.0.0",
    "react-native": "0.72.5",
    "react-native-confetti-cannon": "^1.5.2",
    "react-native-countdown-circle-timer": "^3.1.0",
    "react-native-countdown-component": "^2.7.1",
    "react-native-flash-message": "^0.4.2",
    "react-native-gesture-handler": "~2.12.0",
    "react-native-image-viewing": "^0.2.2",
    "react-native-keyboard-aware-scroll-view": "^0.9.5",
    "react-native-linear-gradient": "^2.6.2",
    "react-native-reanimated": "~3.3.0",
    "react-native-render-html": "^6.3.4",
    "react-native-safe-area-context": "4.6.3",
    "react-native-screens": "~3.22.0",
    "react-native-sound": "^0.11.2",
    "react-native-svg": "13.9.0",
    "react-native-swipe-list-view": "^3.2.9",
    "react-native-webview": "13.2.2",
    "react-native-youtube": "^2.0.2",
    "react-native-youtube-iframe": "^2.3.0",
    "react-redux": "^8.0.2",
    "redux": "^4.2.0",
    "redux-logger": "^3.0.6",
    "redux-thunk": "^2.4.1",
    "remote-redux-devtools": "^0.5.16",
    "yup": "^1.2.0"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0"
  },
  "private": true
}

**Comment by basim-usman** [comment]
Are you able to resolve it ?

**Comment by MatiPl01** [comment]
Hey!

I don't think this issue is related to the `react-native-reanimated` library. The error says that `AppRegistry.registerComponent` was not called. This function is normally used in the main file of your application (`index.js`/`index.ts`) to register your application root component.

Can you provide a repro of the issue?

**Comment by MOHAMED-LAAGUILI** [comment]
@rickyanwar 
did resolve it ? 

cause i running into same issue
expo app crash on open with :

 ERROR cannot read propertry 'prototype' of undefined


@MatiPl01 
@basim-usman

**Reply by basim-usman** [reply]
@MOHAMED-LAAGUILI  i did upgrade my react native version by using react-native-helper and then fix the node_modules libraries competibility then it get working

**Reply by MOHAMED-LAAGUILI** [reply]
@basim-usman  pls would u pls provid more details and steps

**Reply by basim-usman** [reply]
@MOHAMED-LAAGUILI 
Please follow this

1. Please upgrade your react native version according to [](https://react-native-community.github.io/upgrade-helper/)
2.After upgrading the try to Build your app 
3. Resolve all your build errors based upon the node_modules compatibility to meet the requirements of the react native version

**Reply by MatiPl01** [reply]
Hey @MOHAMED-LAAGUILI! Could you please share more details about your setup? Which versions of libraries do you use (expo SDK version, reanimated version, worklets version (if you are using reanimated v4), react native version, etc.)? Ideally, you could create a reproduction repository in which this issue can be reproduced. I believe this issue might be caused by the usage of incompatible versions of libraries but I can't tell for sure without being able to reproduce it on my side.
